[DRAFT] [DO NOT MERGE] Ensure mountain cart can be accessed through a similar interface as its Python Impementation

urmzd opened the pull request:
# Purpose
Provide users the ability to interact with the `rust` implementation as they would with the `python` implementation.

# Changes
- Create a `space` module 
- Make all properties of `MountainCartEnv` public with the exception of the implementation details specific to `rust` (such as `rng`)

# Verification

TODO

urmzd commented:
@MathisWellmann the PR is ready if you'd like to take a look :)

MathisWellmann commented:
Hi,
I appreciate the effort put into this PR and I'll take a more in depth look into all the changes during the coming days.
Initially I see a lot of undocumented code marked with "TODO". That would have to be resolved. The initial lib.rs file had the line "#![deny(missing_docs, missing_crate_level_docs)]" which needs to be put back as I'd like clean docs when its eventually published to crates.io. It looks like there are some unused dependencies as well, which you can check with "cargo +nightly udeps --all-targets", those can be removed. It would be cool if you could add a dependency section in the README to tell users they want to install "sdl2" and "sdl2_gfx", at least on arch, because otherwise the tests won't run. I'm not sure if this is required when just importing this as a library, if not then those dependencies can be downgraded to be "[dev-dependencies]". And finally I feel like having an examples folder with both mountain_car and cartpole is always a good entry point for users to explore a library, so if you could integrate the tests that render those into an examples folder that would be great. Now I'd like to emphasize that this PR has big potential and I'm looking forward to seeing those minor points resolved so we can get this thing merged. Thanks for this work, it's much appreciated :)

urmzd commented:
Perfect, I'll make a list here and check it off as I go on. I'd really like if this library enabled other rustaceans to start ML work easily so I'l be addressing all of the above issues in the weeks to follow :)

urmzd commented:
Hey @MathisWellmann, regarding the documentation, should we try creating new documentation similar to those found in the native python implementation or should we be copying the documentation over to prevent confusion?

urmzd commented:
@MathisWellmann still working on it! Just took a break so nothing has changed since we first chatted. Will ping you as soon as its ready.

urmzd commented:
@MathisWellmann I've documented the source code and updated the README to reflect the dependencies required to interact with the library. I still feel like the source code documentation can be improved but the documentation can be extended in future PR's, what do you think?

urmzd commented:
@MathisWellmann pinging back in-case you missed the first notification :)

MathisWellmann commented:
Thank you for your excellent work!

urmzd commented:
Awesome, looking forward to seeing how these changes impact the gym-rs experience :) There's still a few more things I'd like to contribute. Will open a ticket to start a discussion around future changes!
